components/datetime/src/fields/symbols.rs

```diff
@@ -434,36 +478,86 @@ impl LengthType for Month {
     }
 }
 
-field_type!(Day; {
-    'd' => DayOfMonth = 0,
-    'D' => DayOfYear = 1,
-    'F' => DayOfWeekInMonth = 2,
-    'g' => ModifiedJulianDay = 3,
-}; Numeric; DayULE);
-
-field_type!(Hour; {
-    'K' => H11 = 0,
-    'h' => H12 = 1,
-    'H' => H23 = 2,
-    'k' => H24 = 3,
-}; Numeric; HourULE);
-
-field_type!(Second; {
-    's' => Second = 0,
-    'S' => FractionalSecond = 1,
-    'A' => Millisecond = 2,
-}; Numeric; SecondULE);
-
-field_type!(Week; {
-    'w' => WeekOfYear = 0,
-    'W' => WeekOfMonth = 1,
-}; Numeric; WeekULE);
-
-field_type!(Weekday; {
-    'E' => Format = 0,
-    'e' => Local = 1,
-    'c' => StandAlone = 2,
-}; WeekdayULE);
+field_type!(
+    /// An enum for the possible symbols of a day field in a date pattern.
+    Day; {
+        /// Field symbol for day of month (numeric).
+        'd' => DayOfMonth = 0,
+        /// Field symbol for day of year (numeric).
+        'D' => DayOfYear = 1,
+        /// Field symbol for the day of week occurrence relative to the month (numeric).
+        ///
+        /// For the example `"2nd Wed in July"`, this field would provide `"2"`.  Should likely be paired with the [`Weekday`] field.
+        'F' => DayOfWeekInMonth = 2,
+        /// Field symbol for the modified Julian day (numeric).
+        ///
+        /// The value of this field differs from the conventional Julian day number in a couple of ways, which are based on measuring relative to the local time zone.
+        'g' => ModifiedJulianDay = 3,
+    };
+    Numeric;
+    DayULE
+);
+
+field_type!(
+    /// An enum for the possible symbols of an hour field in a date pattern.
+    Hour; {
+        /// Field symbol for numeric hour [0-11].
+        'K' => H11 = 0,
+        /// Field symbol for numeric hour [1-12].
+        'h' => H12 = 1,
+        /// Field symbol for numeric hour [0-23].
+        'H' => H23 = 2,
+        /// Field symbol for numeric hour [1-24].
+        'k' => H24 = 3,
+    };
+    Numeric;
+    HourULE
+);
+
+field_type!(
+    /// An enum for the possible symbols of a second field in a date pattern.
+    Second; {
+        /// Field symbol for second (numeric).
+        's' => Second = 0,
+        /// Field symbol for fractional second (numeric).
+        ///
+        /// Truncates, like other numeric time fields, but in this case to the number of digits specified by the field length.
+        'S' => FractionalSecond = 1,
```

Comment:
I think this isn't the spot to discuss the truncation behavior

```suggestion
        /// Produces the number of digits specified by the field length.
        'S' => FractionalSecond = 1,
```